sqlgen/db.go

```diff
@@ -361,6 +361,30 @@ func (db *DB) InsertRow(ctx context.Context, row interface{}) (sql.Result, error
 	return db.execWithTrace(ctx, query, "InsertRow")
 }
 
+// InsertRows inserts multiple rows into the database
+//
+// rows should be an array of pointers to a struct, for example:
+//
+//   user1 := &User{Name: "foo"}
+//	 user2 := &User{Name: "fan"}
+//   if err := db.InsertRow(ctx, []{interface{}}{user1, user2}); err != nil {
+//
+func (db *DB) InsertRows(ctx context.Context, rows []interface{}) (sql.Result, error) {
```

Comment:
one minor concern i have is i believe bulk insert is bound by max_allowed_packet configuration which according to this [post](https://aws.amazon.com/blogs/database/best-practices-for-amazon-aurora-mysql-database-configuration/) is 4MB by default on aurora mysql. having this API makes it more likely to accidentally hit this limit and likely we won't uncover this bug until we hit production as local dataset might not be large enough to reflect large customer experience.

therefore, we probably want to always chunk the input to InsertRows to make it safer. it's still not fool-proof since rows can be arbitrarily wide. perhaps there's a way to enforce chunking? what if this API takes a chunk size parameter?

Reply:
regardless of implementation there will always be a limit on batch size though. because the limit of 4MB is so small I worry this will be a recurring source of outage.

Reply:
I noticed the db migration API where we limited batch upserts to 250 as we think that's the optimal number for whatever reasons. What do you think about enforcing this hard limit of 250 on num rows. This is not foolproof either but should catch most cases @changpingc

Reply:
imposing a limit sounds great, but i'd like to make that limit very visible to callers due to the risk of outage. i don't want people to call this without knowing the limit and i claim most people don't read comment when the API looks straightforward. 

what do you think about adding a chunk size parameter and apply chunking here? i think it's great because we'd have to add chunking logic to every caller anyway so this reduces code duplication, and makes it strictly safer.

Reply:
makes sense. Addressed

Reply:
@changpingc